torch_geometric/loader/node_loader.py

```diff
@@ -160,25 +152,19 @@ def filter_fn(
 
         if isinstance(out, SamplerOutput):
             if isinstance(self.data, Data):
-                data = filter_data(
-                    self.data,
-                    out.node,
-                    out.row,
-                    out.col,
-                    out.edge,
-                    self.node_sampler.edge_permutation,
-                )
+                data = filter_data(  #
+                    self.data, out.node, out.row, out.col, out.edge,
+                    self.node_sampler.edge_permutation)
+
             else:  # Tuple[FeatureStore, GraphStore]
-                data = Data(
```

Comment:
@rusty1s I think this code was to add the features to the data.

Reply:
Where was `metadata[-3]` and `metadata[-2]` populated? The sampler does not have access to node features, so I got confused by this part and dropped it.

Reply:
This info is added to the SamplerOutput metadata after sampling and collecting features from all machines in the:
[DistSampler _collate()](https://github.com/pyg-team/pytorch_geometric/blob/9f7e82442efb41687e60469f1e63b0efaa793165/torch_geometric/distributed/dist_neighbor_sampler.py#L717C1-L717C1)

Reply:
The output of `collate_fn()` includes `nfeats, nlabels, efeats` which are our `metadata[-3]`, `metadata[-2]` and `metadata[-1]`. Here: [L717C1-L718C22](https://github.com/pyg-team/pytorch_geometric/blob/88d7986b6d0a6de5895872270d2ff4fc95fae3b7/torch_geometric/distributed/dist_neighbor_sampler.py#L717C1-L718C22)
I will add a small PR to revert these lines.

Reply:
@rusty1s It is different from single node case. for distributed after node sampling there will need one step to put labels, nfeats, efeats from different nodes into pyg data format for loader.